moban/engine.py

```diff
@@ -129,17 +128,29 @@ def _render_with_finding_data_first(self, data_file_index):
                 self.__file_count += 1
 
     def _apply_template(self, template, data, output):
+        temp_dir = ''
```

Comment:
please put this part of code outside as a function, and then write a unit test on the your new function.